### ISSUE TITLE
LoadLocation Call in time package

### ISSUE THREAD

**Issue by rusco** [gopherjs/gopherjs#64]
``` go
package main

import (
    "fmt"
    "math"
    "time"
)

func calcDuration(hrStart, minStart, hrEnd, minEnd int) (hrDiff, minDiff int) {

    // next line crashes on Win7/go1.3, systemcalls not supported:
    //pt, _ := time.LoadLocation("Europe/Lisbon")

    //this works fine:
    pt := time.UTC

    yr, mo, dy := time.Now().Date()

    t1 := time.Date(yr, mo, dy, hrStart, minStart, 0, 0, pt)
    t2 := time.Date(yr, mo, dy, hrEnd, minEnd, 0, 0, pt)

    overallDiff := t2.Sub(t1).Minutes()
    hrDiff = int(math.Floor(overallDiff / 60))
    minDiff = int(math.Mod(overallDiff, 60))

    return
}

func main() {

    hr, mi := calcDuration(8, 15, 19, 35)
    fmt.Printf("Duration: %d hours and %d minutes.\n", hr, mi)
}
```

the "time.LoadLocation" call causes a System call, at least on Windows. 
I don't know how hard it is to implement this, but could you at least
make a hint in the compatibility doc ?

**Comment by MaVo159** [comment]
Since nothing has changed on this front in more than 2 years, maybe it is time to just include zoneinfo.zip. Maybe only if time is imported or even only if LoadLocation is called. I am not sure if the latter is doable, but the former should be. The compilation results are already pretty large (even a small demo is typically >100k compressed), so conditionally adding 300k doesn't sound completely crazy to me.
This issue is quite debilitating given that there is currently no obvious way to manually include timezone 
information besides some really awkward "unsafe" hackery.

**Comment by flimzy** [comment]
It might be that by using an alternate compression algorithm might allow us to reduce the size burden somewhat.  On my system, zoneinfo.zip is 360k. By recompressing with 'zip -9', it is reduced to 300k.  By using gzip (on the original. zip), it is reduced to 96k.

I'm not sure what the best combination would be, for size/performance, but I don't think we _need_ to be stuck with adding 360k to a bundle just to use timezones.

**Comment by MaVo159** [comment]
I didn't think of that. I tried a bunch of combinations (with the decompressed data, then tar, then compression). Looks like around 100k is the lower end with the stuff that is supported directly in go (gzip -9). With xz or 7z it goes down to 50k. Compressing the zip instead of a tar adds 10% for xz, but maybe that is worth it to avoid touching the internals of LoadLocation.

tldr: Using xz on the zip gives 55k.

Leaving this to transport compression would be convenient, but is not an option I guess, since that'll be bad for https.

**Comment by flimzy** [comment]
I'm not sure how lightweight the decompressor for xz is compared to gz, zip, etc. If it's possible to use a browser's built-in gzip decompressor, then using gz offers a huge advantage. But I'm not sure how realistic that is.

**Comment by shurcooL** [comment]
I'll reopen this issue for better visibility since there's renewed interest in resolving this in some way. But we still need to find an acceptable solution, and I'm not sure if adding 100 KB or more (if it needs to also include decompression code) is acceptable.

**Comment by duckbrain** [comment]
Go now has built-in support for embedding the time zone database statically from within in the standard library.

https://golang.org/pkg/time/tzdata/

This might just work as-is (haven't checked), or could be used as a trigger the same way in GopherJS to tell it to include the optimized moment db.

Browser support for IANA timezones has also gotten better. I'm not sure if the browser exposed APIs now cover the needs of `Location`.

https://caniuse.com/?search=Timezone

**Comment by duckbrain** [comment]
Related: #989
